js/model/Presentation.js

```diff
@@ -357,7 +357,17 @@ export var Presentation = {
 
     get title() {
         var svgTitles = this.document.root.getElementsByTagNameNS(SVG_NS, "title");
-        return svgTitles.length ? svgTitles[0].firstChild.wholeText.trim() : "Untitled";
+        if (svgTitles.length) {
+            return svgTitles[0].firstChild.wholeText.trim();
+        }
+        else {
+            var svgRoot = this.document.root.getElementsByTagNameNS(SVG_NS, "svg");
+            if (svgRoot.length) {
+                var docName = svgRoot.firstChild.getAttributeNode("sodipodi:docname").value.trim();
```

Comment:
This part assumes that the document was generated with Inkscape.
We should return "Untitled" if the attribute `sodipodi:docname` does not exist.